taichi/transforms/alg_simp.cpp

```diff
@@ -77,16 +77,7 @@ class AlgSimp : public BasicStmtVisitor {
 
   static void run(IRNode *node) {
     AlgSimp simplifier;
-    while (true) {
```

Comment:
Performance is good but this part is not performance sensitive. Modifying too much can be a bit aggressive. Let's modify only one stmt at a time, otherwise you may crash your program:
https://github.com/taichi-dev/taichi/blob/97dbf64f735598e64dc13690e5f237dedf20f091/taichi/statements.h#L260